website/pages/docs/theming/scale-out-theme.mdx

```diff
@@ -0,0 +1,519 @@
+
+- `fonts` (font families)
+- `fontSizes`
+- `letterSpacings`
+- `lineHeights`
+- `fontWeights`
+
+```ts live=false
+const typography = {
+  fonts: {
+    body: "system-ui, sans-serif",
+    heading: "Georgia, serif",
+    mono: "Menlo, monospace",
+  },
+
+  fontSizes: {
+    xs: "0.75rem",
+    sm: "0.875rem",
+    md: "1rem",
+    lg: "1.125rem",
+    xl: "1.25rem",
+    "2xl": "1.5rem",
+    // More...
+  },
+
+  letterSpacings: {
+    tighter: "-0.06em",
+    tight: "-0.03em",
+    normal: "0",
+    wide: "0.03em",
+    wider: "0.06em",
+    // More...
+  },
+
+  lineHeights: {
+    shorter: 1.3,
+    short: 1.5,
+    base: 1.7,
+    tall: 1.9,
+    taller: "2",
+    // More...
+  },
+
+  fontWeights: {
+    // Custom in special cases. Like this:
+    // normal: 400,
+    // bold: 700,
+  },
+
+}
+
+export type Typography = typeof typography
+
+export default typography
+```
+
+### 💨 Transition
+You can customize transition like this:
+
+```ts live=false
+// transition.ts
+
+const transitionProperty = {
+  common:
+    "background-color, border-color, color, fill, stroke, opacity, box-shadow, transform",
+  colors: "background-color, border-color, color, fill, stroke",
+  dimensions: "width, height",
+  position: "left, right, top, bottom",
+  background: "background-color, background-image, background-position",
+}
+
+const transitionTimingFunction = {
+  "linear": "cubic-bezier(0, 0, 1, 1)",
+  "ease-in": "cubic-bezier(0.42, 0, 1, 1)",
+  "ease-out": "cubic-bezier(0, 0, 0.58, 1)",
+  "ease-in-out": "cubic-bezier(0.42, 0, 0.58, 1)",
+  // More...
+}
+
+const transitionDuration = {
+  "custom-speed": "250ms",
+  // More ...
+}
+
+const transition = {
+  property: transitionProperty,
+  easing: transitionTimingFunction,
+  duration: transitionDuration,
+}
+
+export default transition
+```
+
+
+### 🗂 Z-index
+We recommend trying to solve stacking issues without `z-index`.
+You can learn more about [stacking contexts and z-index here](/guides/z-index).
+
+```ts live=false
+const zIndices = {
+  headerNavigation: 1750,
+  // More...
+}
+
+export type ZIndices = typeof zIndices
+
+export default zIndices
+```
+
+### 📦 Export Foundations
+Export all above overwritten Foundations in `index.ts`. Put this file under `/foundations` folder.
+
+```ts live=false
+// index.ts
+
+import borders from "./borders"
+import breakpoints from "./breakpoints"
+import colors from "./colors"
+import radii from "./radius"
+import shadows from "./shadows"
+import sizes from "./sizes"
+import { spacing } from "./spacing"
+import transition from "./transition"
+import typography from "./typography"
+import zIndices from "./z-index"
+
+const foundations = {
+  breakpoints,
+  zIndices,
+  radii,
+  colors,
+  ...typography,
+  sizes,
+  shadows,
+  space: spacing,
+  borders,
+  transition,
+}
+
+type FoundationsType = typeof foundations
+export interface Foundations extends FoundationsType {}
```

Comment:
Please remove any additional TypeScript type when it is not clear for what it should be used.